MapboxCoreNavigation/NavigationSettings.swift

```diff
@@ -0,0 +1,85 @@
+import Foundation
+
+/**
+ Emitted when something changed in the `NavigationSettings`.
+ */
+public let NavigationSettingsDidChange = Notification.Name("MBNavigationSettingsDidChange")
```

Comment:
Consider using an extension here. 

```swift
extension Notification.Name
{
    static let navigationSettingsDidChange = "MBNavigationSettingsDidChange"
}
```

Reply:
I was following the same pattern we use for route controller notifications b/c `Notification.Name` doesn't bridge to Objective-C, but after a second thought, there's no reason why we shouldn't have properly typed notifications in Swift.